code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -228,6 +232,21 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 
 	return removed
 
+/datum/gas_mixture/remove_specific(gas_id, amount)
+	var/list/cached_gases = gases
+	amount = min(amount, cached_gases[gas_id][MOLES])
+	if(amount <= 0)
+		return null
+	var/datum/gas_mixture/removed = new type
+	var/list/removed_gases = removed.gases
+	removed.temperature = temperature
+	ADD_GAS(gas_id, removed.gases)
+	removed_gases[gas_id][MOLES] = amount
+	cached_gases[gas_id][MOLES] -= amount
+
+	garbage_collect()

```

Comment:
Since this used only to one gas better call garbage_collect() only to one gas.
```
	garbage_collect(list(gas_id))
```

https://github.com/tgstation/tgstation/blob/8e8de7405fd2578f09603e8b3cdb799eb619d27e/code/modules/atmospherics/gasmixtures/gas_mixture.dm#L65-L74